### DISCUSSION TITLE
Extensions and Hooks don't exist in 9.7.1

### DISCUSSION THREAD

**Discussion by Autr** [directus/directus#12560] [Archive]
Hello

1. I've inited a basic hello world extension:  `npm init directus-extension`
2. Edited and confirmed `directus/extensions/endpoints/hello-world/src/index.js` exists
3. `npm run build`
4. Confirmed `directus/extensions/endpoints/hello-world/dist/index.js` exists
5. Run server:

- GET **/extensions/hooks** = _Route /hooks doesn't exist_
- GET **/extensions/endpoints** = _Route /endpoints doesn't exist_
- GET **/extensions/displays** = []
- GET **/extensions/interfaces** = []
- GET **/extensions/layouts** = []
- GET **/extensions/modules** = []
- GET **/extensions/panels** = []

### To Reproduce

`npm init directus-extension`

### Errors Shown

`{"errors":[{"message":"Route /hooks doesn't exist.","extensions":{"code":"ROUTE_NOT_FOUND"}}]}`

<details>
<summary>More Info</summary>

### What version of Directus are you using?

9.7.1

### What version of Node.js are you using?

v16

### What database are you using?

Postgres

### What browser are you using?

Chrome

### What operating system are you using?

MacOS

### How are you deploying Directus?

Ubuntu

</details>

**Comment by Autr** [comment]
Worked it out - the built CommonJS index needs to be in the root of the extension folder rather than the list folder. 

Temporary fix was:

```json
// in package.json
		"build": "rm ./index.js && directus-extension build && mv dist/index.js ./index.js"

```

**Comment by azrikahar** [comment]
> directus/extensions/endpoints/hello-world/dist/index.js

~~The built index.js file should be at the "root", which should be `directus/extensions/endpoints/hello-world/index.js`. Would mind trying that instead?~~ seems like this has been sorted out 👍

> - I've inited a basic hello world extension: npm init directus-extension
> - Edited and confirmed directus/extensions/endpoints/hello-world/src/index.js exists
> - npm run build

If you are running `npm init directus-extension` _inside_ `directus/extensions`, then it is actually not meant like that. Understandably this might be a misled or lack there of in the docs, if that is the case, feedback on how to improve it is definitely welcome!

We should run `npm init directus-extension` _outside_ of the directus folder. Once everything is ok, you can then run `npm run build` and move _only_ the `index.js` file into `/extensions/<extension-type>/<name-of-your-extension>/index.js`.

---

> - GET **/extensions/hooks** = _Route /hooks doesn't exist_
- GET **/extensions/endpoints** = _Route /endpoints doesn't exist_
- GET **/extensions/displays** = []
- GET **/extensions/interfaces** = []
- GET **/extensions/layouts** = []
- GET **/extensions/modules** = []
- GET **/extensions/panels** = []

~~I believe these shouldn't work as GET requests in the first place (especially hooks since they are not actual endpoints),~~ but perhaps they are meant for future usage and still under construction, cc @nickrum to be sure here.

EDIT: scratch that, they can be listed here indeed: 

https://github.com/directus/directus/blob/703fb8426f59619a4f21e80f670ed4347d01666c/api/src/controllers/extensions.ts#L11-L31

**Reply by nickrum** [reply]
Those `/extensions/<type>` endpoints are returning lists of extension names by type. They were used by the App before we started bundling App extensions into a single entrypoint. That's why there are no endpoints for API extensions. I didn't remove those back then because I thought they might still be useful.
